probe_src/probe_py/analysis.py

```diff
@@ -115,8 +115,8 @@ def get_last_pthread(pid, exid, target_pthread_id):
     for pid, process in process_tree_prov_log.processes.items():
         for exec_epoch_no, exec_epoch in process.exec_epochs.items():
             for tid, thread in exec_epoch.threads.items():
-                if tid != 0:
```

Comment:
Fixed a bug by ensuring a thread is not the main thread using `pid` instead of `tid != 0`